src/work/test/WorkTests.cpp

```diff
@@ -812,3 +813,116 @@ TEST_CASE("Work batching", "[batching][work]")
         }
     }
 }
+
+// ======= ConditionalWork tests ======== //
+TEST_CASE("ConditionalWork test", "[work]")
+{
+    VirtualClock clock;
+    Config const& cfg = getTestConfig();
+    Application::pointer appPtr = createTestApplication(clock, cfg);
+    auto& wm = appPtr->getWorkScheduler();
+
+    SECTION("condition satisfied")
+    {
+        auto success = []() { return true; };
+        auto w = std::make_shared<TestBasicWork>(*appPtr, "conditioned-work");
+        wm.executeWork<ConditionalWork>("condition-success", success, w);
+        REQUIRE(w->getState() == BasicWork::State::WORK_SUCCESS);
+    }
+    SECTION("condition failed")
+    {
+        auto parent = wm.scheduleWork<TestWork>("parent-work");
+        auto failedWork =
+            parent->addTestWork<TestBasicWork>("other-work",
+                                               /* will fail */ true, 100);
+        auto condition = [&]() {
+            return failedWork->getState() == TestBasicWork::State::WORK_SUCCESS;
+        };
+
+        auto dependentWork =
+            std::make_shared<TestBasicWork>(*appPtr, "dependent-work");
+        auto conditionedWork = parent->addTestWork<ConditionalWork>(
+            "condition-fail", condition, dependentWork);
+
+        while (!wm.allChildrenDone())
+        {
+            clock.crank();
+        }
+
+        REQUIRE(failedWork->getState() == BasicWork::State::WORK_FAILURE);
+        // Conditioned work was aborted
+        REQUIRE(conditionedWork->getState() == BasicWork::State::WORK_ABORTED);
+        // Dependent work was never started (internally, in PENDING state,
+        // enforced by its destructor)
```

Comment:
I think you mean constructor here?

Reply:
No, I mean destructor, which asserts that BasicWork is in proper state.